test/support/shared_examples/on_duplicate_key_update.rb

```diff
@@ -0,0 +1,126 @@
+def should_support_basic_on_duplicate_key_update
+  describe "#import" do
+    extend ActiveSupport::TestCase::ImportAssertions
+
+    macro(:perform_import){ raise "supply your own #perform_import in a context below" }
+    macro(:updated_topic){ Topic.find(@topic.id) }
+
+    context "with :on_duplicate_key_update an array" do
+      asssertion_group(:should_support_on_duplicate_key_update) do
+        should_not_update_fields_not_mentioned
+        should_update_foreign_keys
+        should_not_update_created_at_on_timestamp_columns
+        should_update_updated_at_on_timestamp_columns
+      end
+
+      context "given columns and values with :validation checks turned off" do
+        let(:columns){  %w( id title author_name author_email_address parent_id ) }
+        let(:values){ [ [ 99, "Book", "John Doe", "john@doe.com", 17 ] ] }
+        let(:updated_values){ [ [ 99, "Book - 2nd Edition", "Author Should Not Change", "johndoe@example.com", 57 ] ] }
+
+        macro(:perform_import) do |*opts|
+          Topic.import columns, updated_values, opts.extract_options!.merge(:on_duplicate_key_update => update_columns , :validate => false)
+        end
+
+        setup do
+          Topic.import columns, values, :validate => false
+          @topic = Topic.find 99
+        end
+
+        context "using an empty array" do
+          let(:update_columns){ [] }
+          should_not_update_fields_not_mentioned
+          should_update_updated_at_on_timestamp_columns
+        end
+
+        context "using string column names" do
+          let(:update_columns){ [ "title", "author_email_address", "parent_id" ] }
+          should_support_on_duplicate_key_update
+          should_update_fields_mentioned
+        end
+
+        context "using symbol column names" do
+          let(:update_columns){ [ :title, :author_email_address, :parent_id ] }
+          should_support_on_duplicate_key_update
+          should_update_fields_mentioned
+        end
+      end
+
+      context "given array of model instances with :validation checks turned off" do
```

Comment:
@zdennis is there a need to duplicate tests for importing an array of models to validate :on_duplicate_key_update functionality? If not, I was going to skinny these examples down a bit.